sklearn/tests/test_common.py

```diff
@@ -66,8 +67,11 @@ def test_non_meta_estimators():
         if name.startswith("_"):
             continue
         for check in _yield_all_checks(name, Estimator):
-            yield check, name, Estimator
-
+            if issubclass(Estimator, ProjectedGradientNMF):
```

Comment:
You can add a comment above on why this is done, in order to avoid confusion here in the future.